scripts/src/mock_client.ts

```diff
@@ -1754,47 +1757,135 @@ async function checkValidTokenAfterLoginRightAfterLogout() {
 	console.log("OK.\n");
 }
 
+async function checkClientMigration() {
+	console.log("===================================== checkClientMigration =====================================");
+
+	const userId = generateId();
+	const deviceId = generateId();
+	const appClient = new SampleAppClient(SMPL_APP_CONFIG.jwtAddress);
+	const jwt = await appClient.getRegisterJWT(userId, deviceId);
+	const client = await MarketplaceClient.create({ jwt });
+
+	await axios.put(`${process.env.MARKETPLACE_BASE}/v2/applications/${client.appId}/blockchain_version`, {
+		blockchain_version: "2"
+	});
+	localCache.clear();
+
+	const keypair = Keypair.random();
+	await client.updateWallet(keypair.secret());
+
+	const selectedOfferV2 = await getOffer(client, "earn", "tutorial");
+
+	console.log(`requesting order for offer: ${ selectedOfferV2.id }: ${ selectedOfferV2.content.slice(0, 100) }`);
+	const openOrderV2 = await client.createOrder(selectedOfferV2.id);
+	console.log(`got order ${ openOrderV2.id }`);
+
+	const content = JSON.stringify({});
+
+	await client.submitOrder(openOrderV2.id, { content });
+	const orderV2 = await retry(() => client.getOrder(openOrderV2.id), order => order.status === "completed", "order did not turn completed");
+
+	console.log(`completion date: ${ orderV2.completion_date }`);
+	console.log(`got order after submit`, orderV2);
+
+	// change blockchain version to 3
+	const kin3BlockchainVersion: BlockchainVersion = "3";
+	await axios.put(`${process.env.MARKETPLACE_BASE}/v2/applications/${client.appId}/blockchain_version`, {
+		blockchain_version: kin3BlockchainVersion
+	});
+	client.requests.blockchainVersion = kin3BlockchainVersion;
+	localCache.clear();
+	console.log(`${client.appId}'s' blockchain_version after a change =`, (await axios.get(`${process.env.MARKETPLACE_BASE}/v2/applications/${client.appId}/blockchain_version`)).data);
+
+	// shouldMigrate API
+	type AccountMigrationStatus = {
+		should_migrate: boolean;
+		app_blockchain_version: BlockchainVersion
+	};
+	const shouldMigrate: AccountMigrationStatus = (await axios.get(`${process.env.MARKETPLACE_BASE}/v2/migration/info/${client.appId}/${keypair.publicKey()}`)).data;
+	console.log(shouldMigrate);
+	expect(shouldMigrate.should_migrate).toBe(true);
+
+	// await new Promise((res, rej) => {
+	// 	setTimeout(() => { res(); }, 10000);
+	// });
+	// burning wallet
+	const burnStatus = await (client.wallet! as kinjs1.Wallet).burn();
+	console.log("burnStatus", burnStatus);
+	expect(burnStatus).toBe(true);
+
+	console.log((await axios.get(`${process.env.MARKETPLACE_BASE}/v2/migration/info/${client.appId}/${keypair.publicKey()}`)).data);
+
+	const selectedOfferV3 = await getOffer(client, "spend");
+	const couponInfo: CouponInfo = JSON.parse(selectedOfferV3.content);
+
+	expect(couponInfo.amount).toEqual(selectedOfferV3.amount);
```

Comment:
remove anything extra such as verifying coupon amount. This isn't related to the test.